### PR TITLE
fix: excessive records by adding the `MaxEvents` field

### LINKED ISSUE #4076: Couldn't delete HttpChaos because request too large
## Bug Report

**What version of Kubernetes are you using?**

Client Version: v1.24.2
Server Version: v1.24.11-gke.1000

**What version of Chaos Mesh are you using?**

ghcr.io/chaos-mesh/chaos-mesh:v2.5.1

**What did you do? / Minimal Reproducible Example**

I want to delete a HttpChaos CR, but the deletion is hanging over, and the chaos-controller's log shows the target HttpFault resource is very large, with an error message `request too large`, seems like an error message from etcd log. So I guess the issue is the size of HttpChaos is too large to delete it in etcd server. When I show the HttpChaos as yaml format, the `status` field is really very long recording all the history of itself.

**What did you expect to see?**

**What did you see instead?**

**Output of chaosctl**

### PR DESCRIPTION
**Pull request by Gallardot**
## What problem does this PR solve?

Close #4076 

## Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`

## Cherry-pick to release branches (optional)

> This PR should be cherry-picked to the following release branches:

- [x] release-2.7
- [x] release-2.6

### CHANGELOG

> Must include at least one of them.

- [x] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

### Tests

> Must include at least one of them.

- [ ] Unit test
- [ ] E2E test
- [x] Manual test

### Side effects

- [ ] **Breaking backward compatibility**

## DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by liamgib** [comment]
We're currently experiencing this issue for NetworkChaos tests, is there any progress in getting this PR merged?

**Comment by Gallardot** [comment]
> We're currently experiencing this issue for NetworkChaos tests, is there any progress in getting this PR merged?

@STRRL @cwen0 @YangKeao @g1eny0ung @xlgao-zju   PTAL

**Comment by g1eny0ung** [comment]
I can take a look this afternoon.

File: .github/workflows/integration_test.yml [hunk 1]
```diff
@@ -112,7 +112,7 @@ jobs:
           helm repo add localstack-repo http://helm.localstack.cloud
           helm upgrade --install localstack localstack-repo/localstack --version 0.6.14
           pip install awscli
-          kubectl wait --timeout=60s --for=condition=ready --all pod
+          kubectl wait --timeout=160s --for=condition=ready --all pod
```
Comment (g1eny0ung) [inline .github/workflows/integration_test.yml:115]: I remembered we've already fixed the integration test, so this change could be removed.

File: controllers/utils/chaosdaemon/chaosdaemon.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func (b *ChaosDaemonClientBuilder) FindDaemonIP(ctx context.Context, pod *v1.Pod
 
 	daemonIP := findIPOnEndpointSlice(&endpointSliceList, nodeName)
 	if len(daemonIP) == 0 {
-		return "", errors.Errorf("cannot find daemonIP on node %s in endpointSliceList %v", nodeName, endpointSliceList)
+		return "", errors.Errorf("cannot find daemonIP on node %s ", nodeName)
```
Comment (g1eny0ung) [inline controllers/utils/chaosdaemon/chaosdaemon.go:70]: Please revert this change.

File: CHANGELOG.md [hunk 3]
```diff
@@ -67,6 +67,7 @@ For more information and how-to, see [RFC: Keep A Changelog](https://github.com/
 - Fix TTL configuration from environment variables [#4338](https://github.com/chaos-mesh/chaos-mesh/pull/4338)
 - Fix dashboard panic while replacing query namespace with targetNamespace in namespace scoped mode [#4409](https://github.com/chaos-mesh/chaos-mesh/issues/4409)
 - Fix incorrect mmap args for IOChaos [#3680](https://github.com/chaos-mesh/chaos-mesh/issues/3680)
+- Fix limit the number of events and minify the events [#4402](https://github.com/chaos-mesh/chaos-mesh/pull/4402)
```
Comment (g1eny0ung) [inline CHANGELOG.md:70]:
```suggestion
- Fix excessive records by adding the `MaxEvents` field [#4402](https://github.com/chaos-mesh/chaos-mesh/pull/4402)
```

**Review by g1eny0ung — CHANGES_REQUESTED** [review]
Others are LGTM.

File: controllers/utils/chaosdaemon/chaosdaemon.go (hunk 2 above)
Comment (Gallardot) [reply controllers/utils/chaosdaemon/chaosdaemon.go:70]: Thanks @g1eny0ung . I fixed everything except this one.

In a large k8s cluster, simply outputting this `discoveryv1.EndpointSliceList` would make the log huge. So I removed that output.

WDYT?

Comment (STRRL) [reply controllers/utils/chaosdaemon/chaosdaemon.go:70]: do not print the detail of endpointslicelist looks OK for me, print the node name could be enough for profiling problems.

**Review by g1eny0ung — APPROVED** [review]

**Comment by g1eny0ung** [comment]
I've pushed these changes directly to release-2.6/2.7 branches.
